mat64/index_bound_checks.go

```diff
@@ -63,3 +65,47 @@ func (m *Vector) set(r int, v float64) {
 	}
 	m.mat.Data[r*m.mat.Inc] = v
 }
+
+// At returns the element at row r and column c.
+func (t *Symmetric) At(r, c int) float64 {
+	return t.at(r, c)
+}
+func (t *Symmetric) at(r, c int) float64 {
+	if r >= t.mat.N || r < 0 {
+		panic(ErrRowAccess)
+	}
+	if c >= t.mat.N || c < 0 {
+		panic(ErrColAccess)
+	}
+	if t.mat.Uplo == blas.Upper {
+		if r > c {
+			r, c = c, r
+		}
+		return t.mat.Data[r*t.mat.Stride+c]
+	}
+	if r < c {
+		r, c = c, r
+	}
+	return t.mat.Data[r*t.mat.Stride+c]
+}
+
+// Set sets the element at row r and column c. Set panics if the location is outside
+// the appropriate half of the matrix.
```

Comment:
That is an alternate possible behavior. What worries me is that this satisfies Mutable, but it isn't as mutible as a true Mutable is (whatever that means). For a Dense matrix, 
````
for i := 0; i < rows; i++ {
     for j := 0; j < cols; j++ {
        m.Set(i,j, float64(i*cols + j))
    }
}
````
does exactly what you think it does. For a Symmetric matrix, it does not, and, in fact, if you loop over the rows first and then the columns, you'll get a different answer than if you do it the other way. This feels like a silent bug to me, especially when interfaces get involved. Furthermore, this felt more along the lines of the BLAS routines, where the opposite-triangle elements are never touched.